server/sandbox_run.go

```diff
@@ -80,7 +83,56 @@ func (s *Server) trustedSandbox(req *pb.RunPodSandboxRequest) bool {
 }
 
 func (s *Server) runContainer(container *oci.Container, cgroupParent string) error {
-	if err := s.Runtime().CreateContainer(container, cgroupParent); err != nil {
+	intermediateMountPoint := container.IntermediateMountPoint()
+
+	if intermediateMountPoint == "" {
+		if err := s.Runtime().CreateContainer(container, cgroupParent); err != nil {
+			return err
+		}
+	} else {
+		errc := make(chan error)
+		go func() {
+			var err error
+
+			defer func(err error) {
+				errc <- err
+			}(err)
+			runtime.LockOSThread()
```

Comment:
Shouldn't we call `defer runtime.UnlockOSThread()` ?

Reply:
I did this on purpose.  With golang 1.10 if a goroutine dies without unlocking the OS thread then the OS thread is destroyed.  This is the safe thing to do as we don't want to run with some threads in a different namespace if for some reason the Setns to the previous mount namespace fails.  Alternatively, I can do the Unlock if there are no errors, as the thread was safely switched to the correct namespace.

Reply:
I think we can do the Unlock on no errors as that will be more frequent case and not doing unlock has performance penalties. Thanks!